genienlp/metrics.py

```diff
@@ -224,23 +223,38 @@ def compute_e2e_dialogue_score(greedy, answer, tgt_lang, args, example_ids, cont
     return results
 
 
-def computeSER(greedy, inputs):
+def computeSER(greedy, inputs, task_names):
+    # TODO: how to address multitask evaluation?
+    assert len(task_names) == 1
+    dataset_class = getattr(dialogues, task_names[0].split('_', 1)[0].capitalize())
+    dataset = dataset_class()
+
     act_values = []
     for input in inputs:
         act_values.append(QUOTED_MATCH_REGEX.findall(input))
 
-    return compute_ser(greedy, act_values)
+    return dataset.compute_ser(greedy, act_values)
 
 
-def computeDA_EM(greedy, answer):
+def computeDA_EM(greedy, answer, task_names):
```

Comment:
what is `task_names`? what is its purpose?

Reply:
`task_names` is a list of tasks you want to evaluate - for example to evaluate a model trained on multiple tasks simultaneously.
We pass it here to be able to initialize the correct dataset (e.g. bitod, risawoz, etc.) from dialogues.

Reply:
This way (`task_names[0].split('_', 1)[0].capitalize()`) of going from task name to dataset class can cause bugs down the line, if you change any task or dataset names. Please add a mapping somewhere in dialogue_task.py instead.

Reply:
Ok. I added it to the branch. Will PR with other changes later.